### DISCUSSION TITLE
What are the "Take over" settings

### DISCUSSION THREAD

**Discussion by mjthoraval** [windingwind/zotero-better-notes#938] [Q&A] [answered]
I see these options in the settings:
![image](https://github.com/windingwind/zotero-better-notes/assets/36054098/e3f6a8c9-c943-4e96-bfc3-3e49460f612e)

Could you explain what they are doing?

Zotero 7.0.0-beta.71+03d0ea436 (64-bit)
BN 1.1.4-beta.61
Windows 10

**Comment by windingwind** [comment]
Take over opening note means the note will be opened in BN tab or BN window (used to be the concept of workspace, which is no longer a thing, since you can have multiple note tabs/windows)

Take over note relation means the note items' related field will be always consistent with their link relations (like if there's a link from note A to B, then they have a relation). this is for the note link graph, which is a prototype for now. It will be able to show the relation in your whole library in future updates. It also means you cannot add/remove related **note** to **note** items (you can still relate note to an regular item, though)

**Comment by mjthoraval** [comment]
It is nice to get rid of the concept of workspace.
But it would still be useful to retain the ability to "pin" a note somehow.

I would strongly prefer separating the "linked notes" from the "related notes", as already mentioned [earlier](https://github.com/windingwind/zotero-better-notes/issues/937).
It would be much safer to separate these features.
And you would get much more freedom on what you can do with the "linked notes".
- For examples, separate the notes linking TO the current note from the notes linking FROM the current note.
- Display the notes linking TO the current notes at the bottom of the note, as in Obsidian, with the context of the mention.
- ...

**Reply by windingwind** [reply]
> I would strongly prefer separating the "linked notes" from the "related notes", as already mentioned https://github.com/windingwind/zotero-better-notes/issues/937.

The related field is the easiest way to make it work, though I do want to investigate the possibility of caching the linked relation locally as files (which might be slower than the relation field stored in DB).

The related field has the advantaged of being able to sync across devices, which might be... useful?

I just want to get the new feature out as soon as possible for beta testing so that we can get them ready before the Z7 stable ideally.

> And you would get much more freedom on what you can do with the "linked notes".

These are planned and can be implemented with either a related field or a local cache.

Thanks and looking for further suggestions/discussions.

**Reply by mjthoraval** [reply]
Thank you for the explanations.

> It also means you cannot add/remove related note to note items (you can still relate note to an regular item, though)

Is there any way around this limitation?
I do not know the technical reasons behind this, but it does not feel right to break a Zotero feature and overwrite pre-existing data because of a new feature.

For me, a very important criteria before I use a plugin is to make sure that when I disable the plugin, everything gets back to normal as before using the plugin. This clearly breaks this rule, so I am quite reluctant to go into this direction. I cannot simply test the plugin, because it will affect my library.

I guess I could simply disable this feature, but I would still be interested in testing the new features in the future.

> I do want to investigate the possibility of caching the linked relation locally as files (which might be slower than the relation field stored in DB).

Is it possible to make a separate DB for BN?

If you move Note Links to Related Items, then why not do the same also other for Item Links or PDF Links? Conceptually, they are all part of a potential links graph, but adding them all will very quickly make things messy.

**Reply by windingwind** [reply]
> but it does not feel right to break a Zotero feature and overwrite pre-existing data because of a new feature.

I agree. So probably it will be cached locally, file or DB.

**Reply by windingwind** [reply, marked as answer]
So I've released a 1.1.4-beta.63, where the related field is no longer used for storing note link relations. The data is now stored in the database locally.

**Reply by ulahcherubim** [reply]
I am very pleased to hear that. I will try new state of the plugin.

**Reply by mjthoraval** [reply]
It looks great, thank you very much.
Looking forward to the upcoming developments in this direction!

**Reply by windingwind** [reply]
There's a 1.1.4-beta.64, where new features like inbound links, outbound links, and directed link graphs are available.

Note that the inbound links might not be accurate because the outbound links, which are someone else's inbound links, are not cached until you first open that note.

For any discussions, we can start a new thread.
